pkcs5/src/pbes2/kdf/salt.rs

```diff
@@ -0,0 +1,104 @@
+//! Salt storage buffer which works on heapless `no_std` targets.
+// TODO(tarcieri): use `ArrayVec` when it's available in `core`.
```

Comment:
It might be worth considering making a crate like `arrayvec` or `tinyvec` an optional dependency of `der` so it can implement much of the boilerplate below.

There are a few places we use fixed-sized buffers like this where it would be nice, I believe.